rfcs/react-components/convergence/motion-definition-n-apis.md

```diff
@@ -0,0 +1,999 @@
+
+</details>
+
+<details>
+  <summary>Why not follow `styled-components` / `framer-motion` API?</summary>
+
+Another API approach might be to compose existing components with motion factories, for example:
+
+```tsx
+// ⚠️ This is not proposed API, it's just an example
+
+import { Button } from '@fluentui/react-components';
+import { createAtomElement, atom } from '@fluentui/react-motions-preview';
+
+const FadeEnterDiv = createAtomElement('div', atom.fade.enterSlow);
+const FadeEnterButton = createAtomElement(Button, atom.fade.enterSlow);
+
+function MyComponent() {
+  return (
+    <>
+      <FadeEnterDiv>Hello world!</FadeEnterDiv>
+      <FadeEnterButton>Hello world!</FadeEnterButton>
+    </>
+  );
+}
+```
+
+This approach does not have obvious pros or cons, but it's worth noting that every animated component will have to be composed, for example:
+
+```tsx
+// ⚠️ This is not proposed API, it's just an example
+
+import { Button, Image } from '@fluentui/react-components';
+import { createAtom, createAtomElement, atom } from '@fluentui/react-motions-preview';
+
+const FadeEnter = createAtom(atom.fade.enterSlow);
+
+const FadeEnterButton = createAtomElement(Button, atom.fade.enterSlow);
+const FadeEnterImage = createAtomElement(Image, atom.fade.enterSlow);
+
+function MyComponent() {
+  return (
+    <>
+      <FadeEnter>
+        <div>Hello world!</div>
+      </FadeEnter>
+      <FadeEnter>
+        <Button>Hello world!</Button>
+      </FadeEnter>
+
+      {/* vs */}
+
+      <FadeEnterDiv>Hello world!</FadeEnterDiv>
+      <FadeEnterButton>Hello world!</FadeEnterButton>
+    </>
+  );
+}
+```
+
+</details>
+
+### Using presence motions
+
+Compared to the CSS option, we don't need to apply classes to the element, so factories don't need to control state and won't suffer from the same issues as the CSS option with re-rendering.
+
+> The Web Animations API has its own state machine [by design](https://developer.mozilla.org/en-US/docs/Web/API/Animation/playState), so we can subscribe, for example, to the onfinish event to handle unmount.
+
+<details>
+<summary>Handling unmount using callbacks</summary>
+
+The simple example is below:
+
+```tsx
+function MyComponent(props) {
+  const { visible, motion } = props;
+
+  const [mounted, setMounted] = useState(visible);
+  const elementRef = React.useRef();
+
+  React.useEffect(() => {
+    if (!visible) {
+      const animation = elementRef.current.animate(motion.keyframes, {
+        duration: motion.duration,
+      });
+
+      animation.onfinish = () => {
+        setMounted(false);
+      };
+    }
+  }, [visible, motion]);
+}
+```
+
+</details>
+
+As with `createAtom()`, the factory returns a React component that clones the child element and applies the provided animation to it:
+
+```tsx
+import { createPresence, presence } from '@fluentui/react-motions-preview';
+
+const FadeSlow = createPresence(presence.fade.slow());
+
+function MyComponent() {
+  return (
+    <FadeSlow>
```

Comment:
should the `visible` prop be used in the example? (and others like appear/unmountOnExit)

Reply:
Good catch, indeed. Updated 98e43f5